notify/compat.go

```diff
@@ -65,3 +69,58 @@ func PostableAPITemplatesToTemplateDefinitions(ts []definition.PostableApiTempla
 	}
 	return defs
 }
+
+// ConfigReceiverToIntegrations converts a ConfigReceiver to a list of MimirIntegrationConfig
+func ConfigReceiverToIntegrations(receiver ConfigReceiver) ([]MimirIntegrationConfig, error) {
+	result := make([]MimirIntegrationConfig, 0)
+	receiverVal := reflect.ValueOf(&receiver).Elem()
+	receiverType := receiverVal.Type()
+	for i := 0; i < receiverType.NumField(); i++ {
+		integrationField := receiverType.Field(i)
+		if integrationField.Type.Kind() != reflect.Slice {
+			continue
+		}
+		sliceType := integrationField.Type
+		elemType := sliceType.Elem()
+		sliceVal := receiverVal.Field(i)
+		if sliceVal.Len() == 0 {
+			continue
+		}
+		iType, err := IntegrationTypeFromMimirTypeReflect(elemType)
+		if err != nil {
+			return nil, err
+		}
+
+		sch, ok := GetSchemaForIntegration(iType)
+		if !ok {
+			return nil, fmt.Errorf("cannot find schema by integration type %s", iType)
+		}
+		var version schema.IntegrationSchemaVersion
+		if sch.Type == iType {
+			version, ok = sch.GetVersion(schema.V0mimir1)
+			if !ok {
+				return nil, fmt.Errorf(" integration type %s does not have version %s", iType, schema.V0mimir1)
+			}
+		} else {
+			version, ok = sch.GetVersionByTypeAlias(iType)
+			if !ok {
+				return nil, fmt.Errorf("cannot find schema version by integration type alias %s", iType)
+			}
+		}
+		result = slices.Grow(result, len(result)+sliceVal.Len())
```

Comment:
This should be the following, no?
```suggestion
		result = slices.Grow(result, sliceVal.Len())
```

Reply:
ah, you're right.